### PR TITLE
Keep unpinned blocks for a while before announcing

### PR DESCRIPTION
**Pull request by cecton**
Related to #7

### CODE REVIEW

File: test/parachain/src/service.rs [hunk 1]
```diff
@@ -72,10 +67,12 @@ macro_rules! new_full_start {
 ///
 /// This function blocks until done.
 pub fn run_collator(
-	parachain_config: Configuration,
+	mut parachain_config: Configuration,
 	key: Arc<CollatorPair>,
 	polkadot_config: polkadot_collator::Configuration,
 ) -> sc_service::error::Result<impl AbstractService> {
+	parachain_config.announce_block = false;
```
Comment (bkchr) [inline test/parachain/src/service.rs:74]: This should be moved into some function provided by Cumulus. As people will need to replicate this all the time and we may need to "tweak" the config even more in the future.

File: test/parachain/src/service.rs [hunk 2]
```diff
@@ -15,17 +15,12 @@
...
 use sc_service::{AbstractService, Configuration};
 use sc_finality_grandpa::{FinalityProofProvider as GrandpaFinalityProofProvider, StorageAndProofProvider};
-
 use polkadot_primitives::parachain::CollatorPair;
-
 use cumulus_collator::CollatorBuilder;
-
-use futures::FutureExt;
-
+use futures::prelude::*;
```
Comment (bkchr) [inline test/parachain/src/service.rs:23]: I'm no huge fan of blindly including preludes. (But it is okay :P )

File: network/src/lib.rs [hunk 3]
```diff
@@ -141,3 +148,105 @@ where
...
+						target: "cumulus-network",
+						"previous task that waits for announce block has been canceled",
+					);
+				}
+			}
+		}).into()) {
+			error!(
+				target: "cumulus-network",
+				"Could not spawn a new task to wait for the announce block: {:?}",
+				err
```
Comment (bkchr) [inline network/src/lib.rs:222]:
```suggestion
				err,
```

File: network/src/lib.rs [hunk 4]
```diff
@@ -141,3 +148,105 @@ where
 		Ok(Validation::Success)
 	}
 }
+
+pub struct WaitToAnnounce<Block: BlockT> {
```
Comment (bkchr) [inline network/src/lib.rs:152]: Some docs would be nice.

File: network/src/lib.rs [hunk 5]
```diff
@@ -141,3 +148,105 @@ where
...
+	spawner: Arc<dyn Spawn + Send + Sync>,
+	network: Arc<NetworkService<Block, <Block as BlockT>::Hash>>,
```
Comment (bkchr) [inline network/src/lib.rs:154]:
```suggestion
	network: Arc<NetworkService<Block, Block::Hash>>,
```

File: test/parachain/src/service.rs (hunk 2 above)
Comment (cecton) [reply test/parachain/src/service.rs:23]: :sloth: will fix it :)

File: network/src/lib.rs (hunk 4 above)
Comment (cecton) [reply network/src/lib.rs:152]: Yes! I was waiting for you to validate the implementation first to be sure

**Comment by cecton** [comment]
hmm 🤔 that's annoying, I need to create a fake NetworkService to fix the test

File: collator/src/lib.rs [hunk 6]
```diff
@@ -433,13 +463,15 @@ mod tests {
 		let id = ParaId::from(100);
 		let _ = env_logger::try_init();
 		let spawner = futures::executor::ThreadPool::new().unwrap();
+		let announce_block = |_, _| ();
```
Comment (cecton) [inline collator/src/lib.rs:466]: :rofl:

**Comment by cecton** [comment]
I replaced the NetworkService parameter with a closure that just does `announce_block`. It removed the dependencies on sc_network.

File: network/src/lib.rs (hunk 4 above)
Comment (cecton) [reply network/src/lib.rs:156]: done

File: test/parachain/src/service.rs (hunk 2 above)
Comment (cecton) [reply test/parachain/src/service.rs]: done

File: test/parachain/src/service.rs (hunk 1 above)
Comment (cecton) [reply test/parachain/src/service.rs:74]: Should I create a crate `service` for this?

Comment (cecton) [reply test/parachain/src/service.rs]: Ok as discussed I placed it in the `collator` crate

**Pull request merged**